src/components/links/LinkBase.tsx

```diff
@@ -0,0 +1,152 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import React, { HTMLAttributeAnchorTarget } from 'react';
+import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
+import { OpenNewTabIcon } from '../../common/icons/OpenNewTabIcon';
+
+type RouterLinkPropsRequired = 'to' | 'reloadDocument' | 'state' | 'download' | 'title';
+export interface LinkProps extends Pick<RouterLinkProps, RouterLinkPropsRequired> {
+  children: React.ReactNode;
+  className?: string;
+  shouldBlurAfterClick?: boolean;
+  isDisabled?: boolean;
+  icon?: React.ReactNode;
+  isExternal?: boolean;
+  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
+  shouldPreventDefault?: boolean;
+  hasExternalIcon?: boolean;
+  shouldStopPropagation?: boolean;
+  target?: HTMLAttributeAnchorTarget;
+}
+
+function LinkBaseWithRef(props: LinkProps, ref: React.ForwardedRef<HTMLAnchorElement>) {
+  const {
+    children,
+    shouldBlurAfterClick = false,
+    isDisabled = false,
+    icon,
+    isExternal: isExternalProp = false,
+    onClick,
+    shouldPreventDefault = false,
+    hasExternalIcon = !icon,
+    shouldStopPropagation = false,
+    to,
+    ...rest
+  } = props;
+
+  const toAsString =
+    typeof to === 'string' ? to : `${to.pathname ?? ''}${to.search ?? ''}${to.hash ?? ''}`;
+
+  const isExternal = isExternalProp || toAsString.startsWith('http');
+
+  const handleClick = React.useCallback(
+    (event: React.MouseEvent<HTMLAnchorElement>) => {
+      if (shouldBlurAfterClick) {
+        event.currentTarget.blur();
+      }
+
+      if (shouldPreventDefault || isDisabled) {
+        event.preventDefault();
+      }
+
+      if (shouldStopPropagation) {
+        event.stopPropagation();
+      }
+
+      if (onClick && !isDisabled) {
+        onClick(event);
+      }
+    },
+    [onClick, shouldBlurAfterClick, shouldPreventDefault, shouldStopPropagation, isDisabled],
+  );
+
+  if (isExternal) {
+    return (
+      <a
+        rel="noopener noreferrer"
+        target="_blank"
+        {...rest}
+        href={toAsString}
+        onClick={handleClick}
+        ref={ref}>
+        {icon}
+        {children}
+        {hasExternalIcon && <ExternalIcon className="sw-ml-1" />}
+      </a>
+    );
+  }
+
+  return (
+    <RouterLink ref={ref} {...rest} onClick={handleClick} to={to}>
+      {icon}
+      {children}
+    </RouterLink>
+  );
+}
+
+const ExternalIcon = styled(OpenNewTabIcon)`
+  color: var(--echoes-color-background-accent-default);
+  margin-left: 4px;
+`;
+
+export const LinkBase = React.forwardRef(LinkBaseWithRef);
+
+export const LinkBaseStyled = styled(LinkBase)`
+  color: var(--color);
+  text-decoration-color: var(--color);
+
+  &:visited {
+    color: var(--color);
+  }
+
+  &:hover,
+  &:focus,
+  &:active {
+    color: var(--active);
```

Comment:
We probably want to make a difference between hover and active since we have two different design tokens for them (with the same color for now), what do you think?